spring-xd-messagebus-rabbit/src/main/java/org/springframework/xd/dirt/integration/rabbit/RabbitMessageBus.java

```diff
@@ -673,6 +677,34 @@ public void destroy() {
 		stopBindings();
 	}
 
+	@Override
+	public void storeForManualAck(MessageHeaders headers) {
+		if (headers.containsKey(AmqpHeaders.CHANNEL)) {
+			Channel channel = (com.rabbitmq.client.Channel) headers.get(AmqpHeaders.CHANNEL);
+			Long deliveryTag = (Long) headers.get(AmqpHeaders.DELIVERY_TAG);
+			if (channelsForManualAck.containsKey(channel) && (channelsForManualAck.get(channel) < deliveryTag)) {
+				channelsForManualAck.put(channel, deliveryTag);
+			}
+			else {
+				channelsForManualAck.put(channel, deliveryTag);
+			}
```

Comment:
The code is identical regardless of the if condition.

This is actually correct because you can't assume deliveryTags are incrementing.

Let's say your "batch size" is 10 and you've already pushed 10.

Now we deliver 11-15 and there's a network glitch (connection reset).

11-15 will be redelivered but they will have delivery tags 1-5.